spec/classes/init_spec.rb

```diff
@@ -1,104 +1,107 @@
 require 'spec_helper'
 describe 'inittab' do
-
   describe 'with parameter default_runlevel' do
-    %w{ 0 1 2 3 4 5 6 s S }.each do |runlevel|
+    ['0', '1', '2', '3', '4', '5', '6', 's', 'S'].each do |runlevel|
       context "set to #{runlevel}" do
-        it "should not fail on runlevel #{runlevel}" do
-          expect { runlevel }.to_not raise_error
+        it "is_expected.to not fail on runlevel #{runlevel}" do
+          expect { runlevel }.not_to raise_error
         end
       end
     end
   end
 
   describe 'with default_runlevel set to invalid value' do
-    let(:params) { { :default_runlevel => '10' } }
+    let(:params) { { default_runlevel: '10' } }
+
     let :facts do
-      { :osfamily               => 'RedHat',
-        :operatingsystem        => 'RedHat',
-        :operatingsystemrelease => '6.5',
+      {
+        osfamily:               'RedHat',
+        operatingsystem:        'RedHat',
+        operatingsystemrelease: '6',
       }
     end
 
-    it 'should fail' do
+    it 'is expected to fail' do
       expect {
-        should contain_class('inittab')
-      }.to raise_error(Puppet::Error,/default_runlevel <10> does not match regex/)
```

Comment:
The error messages are part of the testing, else you can have an error for unrelated things, which is not a good test.